File: platforms/ios/Sources/WhiskerRuntime/paint/PaintColor.swift
import UIKit
import WhiskerModule

func parsePaintColor(_ value: WhiskerMobileColor) -> UIColor {
    if value.kind == 1 {
        return UIColor(
            red: CGFloat(value.red) / 255,
            green: CGFloat(value.green) / 255,
            blue: CGFloat(value.blue) / 255,
            alpha: CGFloat(value.alpha)
        )
    }
    switch hostString(value.name).lowercased() {
    case "black": return .black
    case "white": return .white
    case "red": return .red
    // UIKit's `.green` is #00ff00; CSS named `green` is #008000.
    case "green": return UIColor(red: 0, green: 128 / 255, blue: 0, alpha: 1)
    case "gold": return UIColor(red: 1, green: 215 / 255, blue: 0, alpha: 1)
    case "blue": return .blue
    default: return .clear
    }
}

File: platforms/android/runtime/src/main/kotlin/rs/whisker/runtime/paint/PaintColor.kt
package rs.whisker.runtime.paint

import android.graphics.Color

internal fun rgba(red: Float, green: Float, blue: Float, alpha: Float): Int = Color.argb(
    (alpha * 255f).toInt().coerceIn(0, 255),
    red.toInt().coerceIn(0, 255),
    green.toInt().coerceIn(0, 255),
    blue.toInt().coerceIn(0, 255),
)

internal fun parseNamedColor(name: String): Int =
    when (name.lowercase()) {
        "gold" -> Color.rgb(255, 215, 0)
        // Android's named `green` is #00ff00, while CSS defines `green` as
        // #008000. Keep Host paint color resolution aligned with CSS/WPT.
        "green" -> Color.rgb(0, 128, 0)
        "transparent" -> Color.TRANSPARENT
        else -> runCatching { Color.parseColor(name) }.getOrDefault(Color.TRANSPARENT)
    }
